python/pyarrow/_flight.pyx

```diff
@@ -46,6 +46,13 @@ cdef class Action:
     def body(self):
         return pyarrow_wrap_buffer(self.action.body)
 
+    @staticmethod
+    cdef Action unwrap(action):
+        if not isinstance(action, Action):
+            raise TypeError("Must provide Action, not '{}'".format(
+                type(action)))
+        return <Action> action
```

Comment:
This is called `unwrap`, but doesn't actually unwrap (from the name, I would expect it to return e.g. a `CAction*`).